### DISCUSSION TITLE
Only kinda works --

### DISCUSSION THREAD

**Discussion by ghd1492** [andyholmes/gnome-shell-extension-gsconnect#1590] [Q&A]
I've loved this for the last year or so.   A couple of months ago, it started failing ---  Not totally.  But enough to be mostly useless.   it on a variety of machines:  Rocky Linux 9.1 (I have 2)  and Mint 20 (laptop) ..  The failures are consistent on all of them.   My phone is a Samsung Note 20 - Android version 13,  One UI 5.1 (Dilbert is correct about UI designers) Kernel version 4.19.113.-25904254 , Build #TP1A.220624.014.N981USQS3HWB7 -- 
I can see the phone.   I can send files from the pc to the phone.   Sometimes I can see texts.   I cannot send texts to the phone The Troubleshooting file isn't very helpful -- generally only has a couple of lines.. 
I don't think the pc's are messed up because they all seem to exhibit the same problems.    I suspect Samsung / T-Mobile -- some sloppy programmer there.   Has anyone got a clue.   I like this.   My Signal app works fine.  But I'd like to be able to just do regular SMS with this like I used to do.

**Comment by ferdnyc** [comment]
@ghd1492 Text messaging is... very much a crapshoot on Samsung phones, unfortunately. They have their own ideas about how the standard Android SMS database should be implemented, apparently, and the KDE Connect Android project has struggled mightily with getting access to the necessary fields in various phone models' quirky SMS datastores.

See, for example, [this bug search](https://bugs.kde.org/buglist.cgi?component=messaging-application&f1=OP&f2=longdesc&f3=short_desc&f4=longdesc&f5=CP&f6=OP&f7=longdesc&f8=short_desc&j1=OR&j6=OR&o2=substring&o3=substring&o4=allwordssubstr&o7=substring&o8=substring&product=kdeconnect&query_format=advanced&v2=sms&v3=sms&v4=text%20message&v7=samsung&v8=samsung) in the kdeconnect-android bugtracker, which lists only reports that contain "Samsung" along with "SMS" or "text message". (There are 9 entries, and a better search would probably turn up far more.)

All but the most recent two are marked "Fixed" or "Duplicate" (and one of the two open ones is a feature request), but in my experience as someone on their fourth Samsung model, getting any of them to play nice with SMS has been a major headache.

You _could_ try switching on the "Legacy SMS Support" option way at the bottom of **Mobile Settings > $DEVICE > Advanced**, and see if that makes any difference. You might also want to clear the Device Cache (using the button immediately above the Legacy SMS switch), especially if switching to/from legacy mode, as that'll flush out the local cache of SMS data and attempt a resync with the phone.

Be warned that if, like me, you have a significant SMS history on the device — particularly if it contains lots of MMS texts with embedded photos — syncing the SMS history with GSConnect every time it reacquires the phone will be serious drag on the linux <-> Android link. Affecting _everything_, not just SMS.

Text message feeds from Samsung seem to always include embedded image thumbnails directly in the SMS history data, even though GSConnect never asked for them, which makes quick or low-impact syncing pretty much impossible for anyone whose device contains a normal (cluttered, deep) SMS history, unless they want to get draconian and blow it all away. (Even archiving things won't help: [archived and spam-canned texts are also retrieved and sent to the paired system by `kdeconnect-android`](https://bugs.kde.org/show_bug.cgi?id=442756).)
